src/types/boolobject.rs

```diff
@@ -18,6 +20,24 @@ impl PyBool {
     /// Gets whether this boolean is `true`.
     #[inline]
     pub fn is_true(&self) -> bool {
+        Py2::<PyBool>::borrowed_from_gil_ref(&self).is_true()
+    }
+}
+
+/// Implementation of functionality for [`PyBool`].
+///
+/// These methods are defined for the `Py2<'py, PyBool>` smart pointer, so to use method call
+/// syntax these methods are separated into a trait, because stable Rust does not yet support
+/// `arbitrary_self_types`.
```

Comment:
I know I'm late to this party, but why does
```
impl<'py> Py2<'py, PyBool> {
    ...
}
```
not work?

Reply:
IIRC, the main problem with that approach was documentation as this would make all methods for all types end up documentation on single page for the `Py2` type.

Reply:
That, plus we end up leaving `rust_numpy` and other downstream crates out in the cold, as they cannot add methods to `Py2<T>`. This approach works consistently for everyone.

My long-term hope is that arbitrary self types land and we can remove the traits more or less without anyone noticing.

Reply:
:+1: